pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -91,35 +140,236 @@ var _ = Describe("MachineSync Reconciler", func() {
+				mapiMachine = mapiMachineBuilder.Build()
+				Expect(k8sClient.Create(ctx, mapiMachine)).Should(Succeed())
 
-		_, err := reconciler.Reconcile(ctx, reconcile.Request{
-			NamespacedName: types.NamespacedName{
-				Namespace: namespaceName,
-				Name:      machine.Name,
-			},
+				By("Setting the MAPI machine AuthoritativeAPI to MachineAPI")
+				Eventually(k.UpdateStatus(mapiMachine, func() {
+					mapiMachine.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityMachineAPI
+				})).Should(Succeed())
+			})
+
+			Context("when the CAPI machine does not exist", func() {
+				It("should create the CAPI machine", func() {
+					Eventually(k.Get(
+						capiv1resourcebuilder.Machine().WithName(mapiMachine.Name).WithNamespace(capiNamespace.Name).Build(),
+					)).Should(Succeed())
+				})
+
+				It("should update the synchronized condition on the MAPI machine to True", func() {
+					Eventually(k.Object(mapiMachine), timeout).Should(
+						HaveField("Status.Conditions", ContainElement(
+							SatisfyAll(
+								HaveField("Type", Equal(consts.SynchronizedCondition)),
+								HaveField("Status", Equal(corev1.ConditionTrue)),
+								HaveField("Reason", Equal("ResourceSynchronized")),
+								HaveField("Message", Equal("Successfully synchronized MAPI Machine to CAPI")),
+							))),
+					)
+				})
+			})
+
+			Context("when the CAPI machine does exist", func() {
+				BeforeEach(func() {
+					capiMachine = capiMachineBuilder.Build()
+					Expect(k8sClient.Create(ctx, capiMachine)).Should(Succeed())
+				})
+
+				It("should update the synchronized condition on the MAPI machine to True", func() {
+					Eventually(k.Object(mapiMachine), timeout).Should(
+						HaveField("Status.Conditions", ContainElement(
+							SatisfyAll(
+								HaveField("Type", Equal(consts.SynchronizedCondition)),
+								HaveField("Status", Equal(corev1.ConditionTrue)),
+								HaveField("Reason", Equal("ResourceSynchronized")),
+								HaveField("Message", Equal("Successfully synchronized MAPI Machine to CAPI")),
+							))),
+					)
+				})
+			})
+		})
+
+		Context("when the MAPI machine has MachineAuthority set to Migrating", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI and MAPI machines")
+				// We want a difference, so if we try to reconcile either way we
+				// will get a new resourceversion
+				mapiMachine = mapiMachineBuilder.Build()
+				capiMachine = capiMachineBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, capiMachine)).Should(Succeed())
+				Expect(k8sClient.Create(ctx, mapiMachine)).Should(Succeed())
+
+				By("Setting the AuthoritativeAPI to Migrating")
+				Eventually(k.UpdateStatus(mapiMachine, func() {
+					mapiMachine.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityMigrating
+				})).Should(Succeed())
+			})
+
+			It("should not make any changes to either machine", func() {
+				mapiResourceVersion := mapiMachine.GetResourceVersion()
+				capiResourceVersion := capiMachine.GetResourceVersion()
```

Comment:
Might want to add a comment explaining that this is the original RV since we haven't fetched the resource since it was created?

Reply:
Done.